src/legacy/core_plugins/visualizations/public/np_ready/public/components/visualization_chart.tsx

```diff
@@ -63,6 +63,10 @@ class VisualizationChart extends React.Component<VisualizationChartProps> {
           throw new Error('Visualization implementation was not initialized on first render.');
         }
 
+        if (this.visualization.updateVis) {
```

Comment:
Looks like it should help with fixing the issue, but from the app design perspective not sure that it is a best solution.

Reply:
Do you have any other ideas? We need to update vis object inside visualization.

Reply:
using visParams instead of vis.params works as a work around right ?

the problem is that vis is passed to the constructor, render method is passed visParams (which are correct), but vis object that was passed to constructor is no longer up to date.

i think correct solution would be to either:
- pass vis to render method (as it keeps changing)
- preserve vis between renders (so its not changing)

but as we plan to dismantle this whole visualization insfrastructure i think using the workaround for now is good.

Reply:
Yes, Using visParams allows avoiding a problem. 
Do you mean that we should use visParams and close this PR?

Reply:
yes